src/domain/accounts/services/src/authentication_service_impl.rs

```diff
@@ -198,182 +187,353 @@ impl AuthenticationServiceImpl {
         }
         Ok(())
     }
+
+    async fn account_by_id_impl(
+        &self,
+        account_id: &AccountID,
+        account_repository: &Arc<dyn AccountRepository>,
+        maybe_login_password_auth_provider: &Option<Arc<LoginPasswordAuthProvider>>,
+    ) -> Result<Option<Account>, InternalError> {
+        self.ensure_predefined_accounts_registration(
+            account_repository,
+            maybe_login_password_auth_provider,
+        )
+        .await?;
+
+        match account_repository.get_account_by_id(account_id).await {
+            Ok(account) => Ok(Some(account.clone())),
+            Err(GetAccountByIdError::NotFound(_)) => Ok(None),
+            Err(GetAccountByIdError::Internal(e)) => Err(e),
+        }
+    }
+
+    fn get_authentication_providers_by_method(
+        &self,
+        transactional_catalog: &TransactionalCatalog,
+    ) -> HashMap<&'static str, Arc<dyn AuthenticationProvider>> {
+        let authentication_providers = transactional_catalog
+            .get::<AllOf<dyn AuthenticationProvider>>()
+            .unwrap();
+        let mut authentication_providers_by_method = HashMap::new();
+
+        for authentication_provider in authentication_providers {
+            let login_method = authentication_provider.provider_name();
+
+            let insert_result = authentication_providers_by_method
+                .insert(login_method, authentication_provider.clone());
+
+            assert!(
+                insert_result.is_none(),
+                "Duplicate authentication provider for method {login_method}"
+            );
+        }
+
+        authentication_providers_by_method
+    }
 }
 
-///////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
 
 #[async_trait::async_trait]
 impl AuthenticationService for AuthenticationServiceImpl {
-    fn supported_login_methods(&self) -> Vec<&'static str> {
-        self.authentication_providers_by_method
-            .keys()
-            .copied()
-            .collect()
+    async fn supported_login_methods(&self) -> Result<SupportedLoginMethods, InternalError> {
+        self.transaction_runner
+            .transactional(|transactional_catalog| async move {
+                let mut login_methods: Vec<_> = self
+                    .get_authentication_providers_by_method(&transactional_catalog)
+                    .keys()
+                    .copied()
+                    .collect();
+
+                // Stable order for tests
+                login_methods.sort_unstable();
+
+                Ok(login_methods)
+            })
+            .await
```

Comment:
I don't fully understand why this requires a transaction

Reply:
`AuthenticationServiceImpl` no longer manages transactions

---

> I don't fully understand why this requires a transaction

But to answer the question: it was a necessity from a DI perspective